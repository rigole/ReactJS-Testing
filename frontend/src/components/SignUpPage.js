import React  from "react";

function SignUpPage () {
        return (
            <div>
                <h1>Hello world</h1>
                <label>Username</label>
                    <input type="text" placeholder="username"/>
                <label>Email</label>
                    <input type="text" placeholder="email"/>
            </div>
        )
}

export default SignUpPage