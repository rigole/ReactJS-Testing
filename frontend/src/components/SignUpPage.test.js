import SignUpPage from "./SignUpPage";
import '@testing-library/jest-dom'
import { render, screen } from "@testing-library/react"
import userEvent from "@testing-library/user-event";
const sign = require("./SignUpPage")


   /*test("add header", () => {
       expect(sign).toMatchSnapshot("<h1>")
   })*/

describe('Sign Up Page', () => {
    describe('Layout', () => {
        /*it('should have input',  () => {
            //expect(sign).toMatchSnapshot("<input/>")
            const { container  } = render(<SignUpPage/>)
            const input = container.querySelector('input');
            expect(input).toBeInTheDocument();
        });*/
        /*it('has email input', () => {
            const { container } = render(<SignUpPage/>);
            const inputs = container.querySelectorAll('input');
            expect(inputs.length).toBe(2);
        });
        it('has username label', () => {
            render(<SignUpPage/>);
            const input = screen.getByLabelText('username');
            expect(input).toBeInTheDocument();
        });
        it('has email input', () => {
            render(<SignUpPage/>);
            const input = screen.getByPlaceholderText('email');
            expect(input).toBeInTheDocument();
        });*/
    });
     describe("Testing interactions" , () => {
         it("enables the button when password have same savuel" , ()=> {
             render(<SignUpPage/>);
             const passwordInput = screen.getByLabelText('password');
             const passwordRepeat = screen.getByLabelText("Password Repeat")
             userEvent.type(passwordInput, "Password");
             userEvent.type(passwordRepeat, "Password");
             const button = screen.queryByRole('button', { name: 'Sign Up' });
             expect(button).toBeDisabled();
         })
     })
})

