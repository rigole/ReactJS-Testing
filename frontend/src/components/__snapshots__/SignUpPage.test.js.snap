// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add header: <h1> 1`] = `
Object {
  "default": [Function],
}
`;
